src/item.cpp

```diff
@@ -1870,9 +1871,11 @@ nc_color item::color_in_inventory() const
         switch( rating ) {
             case EDIBLE:
             case TOO_FULL:
-                if( is_going_bad() ) {
+                if( preserves ) {
+                    // Nothing, canned food won't rot
```

Comment:
Canned food only really becomes perishable when removed from the can.
In most cases there is no good reason to un-can anything except during eating.
The old coloring was misleading - it hid the fact that cans prevent rot.